### DISCUSSION TITLE
给歌手添加歌曲有问题

### DISCUSSION THREAD

**Discussion by SHJXBJS** [Yin-Hongwei/vue-spring-music#223] [Q&A]
![image](https://user-images.githubusercontent.com/96714686/203251212-d046dd35-abbf-42a6-bb06-87d7a94f0fa0.png)

**Comment by SHJXBJS** [comment]
这是后端的报错
22-11-22 15:20:42,100  WARN DefaultHandlerExceptionResolver:208 - Resolved [org.springframework.web.HttpMediaTypeNotSupportedException: Content type 'multipart/form-data;boundary=----WebKitFormBoundary2EwBrZGhBAvTzZRH;charset=UTF-8' not supported]

**Comment by SHJXBJS** [comment]


**Reply by SHJXBJS** [reply]
> > > 问题所在： ![image](https://user-images.githubusercontent.com/96714686/203256376-791c5d16-9e92-4a85-857b-ff8b7359d696.png) 解决方法：![image](https://user-images.githubusercontent.com/96714686/203257075-c6a89bbd-b63a-4e1d-846e-0d03fecbd112.png) 原因： multipart/form-data 就不能使用 @requestbody，这是因为request的body部分的数据编码格式由header部分的Content-Type指定；
> > > @requestbody作用： 1. 该注解用于读取Request请求的body部分数据，使用系统默认配置的HttpMessageConverter进行解析，然后把相应的数据绑定到要返回的对象上；  2. 再把HttpMessageConverter返回的对象数据绑定到 controller中方法的参数上。

**Reply by Yin-Hongwei** [reply]
😂 哈哈哈，刚看见，我晚上有时间看一下，谢谢指出 👍
